### PR TITLE
improvement(ci): Now github should also check if the nix dev-env/buil…

### PR DESCRIPTION
**Pull request by x10an14**
…d breaks

### CODE REVIEW

File: .github/workflows/nix.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: Ensure nix-dev environment still builds correctly
```
Comment (shadows-withal) [inline .github/workflows/nix.yml:1]: maybe shorten the name so that it doesn't overflow in the interface?

Comment (x10an14) [reply .github/workflows/nix.yml:1]: Sure, got a suggestion I can click "add" on here in review comment? I'm currently away from pc (on smartphone device you see).

Comment (shadows-withal) [reply .github/workflows/nix.yml:1]:
```suggestion
name: Verify nix environment
```

**Comment by x10an14** [comment]
Needs some debugging/fixing as to why the x86_64-linux experience on multiple private machines, break on the Ubuntu Github Actions build envs.

Any help from others to test the nix commands that fail is very welcome.

**Comment by markgreene74** [comment]
hi @x10an14 I gave your workflow a try, and this is what I found:

## nix-build-and-run-rustlings

The [`nix-build-and-run-rustlings`](https://github.com/rust-lang/rustlings/actions/runs/6073126082/job/16474440782?pr=1651#logs) job seems to fail because of the `rustc` version.

Error:

```
> error: package `clap_derive v4.4.2` cannot be built because it requires rustc 1.70.0 or newer, while the currently active rustc version is 1.64.0
```

The error does not happen if I try a new build. I guess the `rustc` version was maybe bumped up in the Nix build.

Here is the link to a successful run with the `--debug` flag: [`nix-build-and-run-rustlings`](https://github.com/markgreene74/rustlings/actions/runs/6597960093/job/17925458202#logs)

```
copying path '/nix/store/66a3ip9miaimi7gyhl6jdzbjchp76ccv-rustc-1.72.0' from 'https://cache.nixos.org/'...
locking path '/nix/store/66a3ip9miaimi7gyhl6jdzbjchp76ccv-rustc-1.72.0'
lock acquired on '/nix/store/66a3ip9miaimi7gyhl6jdzbjchp76ccv-rustc-1.72.0.lock'
```

So this seems OK now.

## nix-dev-shell

The [nix-dev-shell](https://github.com/rust-lang/rustlings/actions/runs/6073126082/job/16474440394?pr=1651#logs) job fails with this error:

```
error: flake output attribute 'devShell' is not a derivation or path
```

I am not very good with Nix but looking at the `flake.nix` file and the Flake docs [here](https://nixos.wiki/wiki/Flakes#Output_schema) and [here](https://nixos.wiki/wiki/Flakes#Importing_packages_from_multiple_channels) the error makes sense, as it doesn't look like `devShell` is defined as `packages."<system>"."<name>"`.

If I replace `nix build .#devShell` with `nix build .#rustlings-app` instead I get this different error:

```
error: flake 'git+file:///home/runner/work/rustlings/rustlings?shallow=1' does not provide attribute 'packages.x86_64-linux.rustlings-app', 'legacyPackages.x86_64-linux.rustlings-app' or 'rustlings-app'
```

I did some additional testing in this branch if anybody is interested in seeing the results: https://github.com/markgreene74/rustlings/tree/test/x10an14-fork-add-nix-github-workflow

**Comment by mo8it** [comment]
Closing because of missing response. Things will change in v6 anyway so we probably need something else then.

**Comment by x10an14** [comment]
Apologies for missing the message you expected a response to. What are the plans for v6? Where/what issue/PR can I read up on them?

**Pull request closed without merging**